### PR TITLE
frontend with all the fixes

### PR DESCRIPTION
**Pull request by ongspxm**
rebased with master.

**commit message**
```
Dashboard doesn't support these features:
- generating of the weekly commit data from daily commit information (#134).
- indicate loading status when loading (#49)
- spacing between filters (#47)

Dashboard is not structured to load in the report information as individual
files. Current structure requires there be multiple template files to be embed
within each other using iframes.

Let's restructure the dashboard into a single SPA which ultilizes vue more
fully. The summary and authorship view are split into different modules to
allow for easier mantainance of the modules in the future as well.
```

### CODE REVIEW

**Comment by yong24s** [comment]
@eugenepeh 

Should we setup checkstyle for JS? `ESLint` supports airbnb style.

**Comment by ongspxm** [comment]
@eugenepeh probably can remove codacy, i gonna add in eslint into travis for the code quality checks.

**Comment by yamidark** [comment]
@ongspxm Could you make the display name on the right panel be the same as the display name in the overview?

**Comment by eugenepeh** [comment]
@ongspxm this requires #152 in order to be fully functional, right?

**Comment by ongspxm** [comment]
Nope. The user can self zip if they use have no means of starting a server.
on the other hand, starting with a server works.

**Comment by eugenepeh** [comment]
> Nope. The user can self zip if they use have no means of starting a server.
on the other hand, starting with a server works.

As discussed, keep server out wherever possible.

**Comment by ongspxm** [comment]
Then need lorh.. need zip for them

File: .travis.yml [hunk 1]
```diff
@@ -6,9 +6,10 @@ matrix:
       jdk: oraclejdk9
     - os: osx
 
-script: >-
-    ./config/travis/run-checks.sh &&
-    time travis_retry ./gradlew clean checkstyleMain checkstyleTest test functional
+script:
+  - ./config/travis/run-checks.sh
+  - cd ./frontend; npm run lint src/static/js/*js; cd ..
```
Comment (eugenepeh) [inline .travis.yml]: why dont just make it frontend/src

File: .travis.yml [hunk 2]
```diff
@@ -20,6 +21,9 @@ before_install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew reinstall --with-pcre2 git; fi
 
+  # install npm tools for eslint
+  - cd ./frontend; npm install --only=dev; cd ..
```
Comment (eugenepeh) [inline .travis.yml]: npm install --only=dev /frontend?

File: .travis.yml (hunk 1 above)
Comment (ongspxm) [reply .travis.yml]: are you referring to the file path? 

Well, eslint is kind of a pesky fellow, on my side, it doesnt recursively lints through all the files, well, it claims it does, but it doesnt, and it says its some bug thingy.

All the JS files are in one place anyways, so i figured, why not?

File: .travis.yml (hunk 2 above)
Comment (ongspxm) [reply .travis.yml]: not sure if that works, let me try.

Comment (ongspxm) [reply .travis.yml]:
```
$ npm install --only=dev /frontend
npm ERR! code ENOLOCAL
npm ERR! Could not install from "../../../../frontend" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ongspxm/.npm/_logs/2018-06-29T03_15_16_215Z-debug.log
```

File: .travis.yml (hunk 1 above)
Comment (eugenepeh) [reply .travis.yml]: i mean, why do you need to cd if you can just append ./frontend in front of src?

File: .travis.yml (hunk 2 above)
Comment (eugenepeh) [reply .travis.yml]: `"../../../../frontend"` seems like its not using correct directory, try ./frontend

**Comment by ongspxm** [comment]
@eugenepeh also updated the branch

**Comment by eugenepeh** [comment]
why is this PR even bigger than before? use merge instead of rebase for this.

**Comment by ongspxm** [comment]
wait... should i rebased to upstream/frontend or upstream/master?

**Comment by ongspxm** [comment]
@eugenepeh this is done, oni merged with upstream/master, i will make another PR to make the rest of the cleaning up changes

**Pull request merged**